scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1015,9 +1015,10 @@ def _create_build_dir_link(self, links_dir_path, instance):
         self.link_dir_counter += 1
 
 
-def change_skip_to_error_if_integration(options, instance):
-    ''' If integration mode is on all skips on integration_platforms are treated as errors.'''
-    if options.integration and instance.platform.name in instance.testsuite.integration_platforms \
+def change_skip_to_error_on_integration_and_allow_platforms(options, instance):
+    ''' All skips on platforms from integration_platforms and platform_allow are treated as errors.'''
```

Comment:
you will need to change the message, the error will talk about integration platforms when it is actually platform_allow, see ```INFO: Error found: tests/arch/arc/arc_dsp_sharing/dsp_sharing.test_calculation on nsim_em11d (Not in testsuite toolchain allow list but is one of the integration platforms)```

Reply:
btw, the above seems to  be a false negative.